temporal-sdk/src/test/java/io/temporal/internal/replay/ReplayWorkflowRunTaskHandlerTaskHandlerTests.java

```diff
@@ -208,6 +210,143 @@ public void ifStickyExecutionAttributesAreSetThenWorkflowsAreCached() throws Thr
     assertEquals(Durations.fromSeconds(5), attributes.getScheduleToStartTimeout());
   }
 
+  @Test
+  public void setsSdkNameAndVersionIfNotSetInHistory() throws Throwable {
+    assumeFalse("skipping for docker tests", SDKTestWorkflowRule.useExternalService);
+
+    WorkflowExecutorCache cache =
+        new WorkflowExecutorCache(10, new WorkflowRunLockManager(), new NoopScope());
+    WorkflowTaskHandler taskHandler =
+        new ReplayWorkflowTaskHandler(
+            "namespace",
+            setUpMockWorkflowFactory(),
+            cache,
+            SingleWorkerOptions.newBuilder().build(),
+            InternalUtils.createStickyTaskQueue("sticky", "taskQueue"),
+            Duration.ofSeconds(5),
+            testWorkflowRule.getWorkflowServiceStubs(),
+            null);
+
+    PollWorkflowTaskQueueResponse workflowTask =
+        HistoryUtils.generateWorkflowTaskWithInitialHistory();
+
+    WorkflowTaskHandler.Result result = taskHandler.handleWorkflowTask(workflowTask);
+
+    assertTrue(result.isCompletionCommand());
+    assertEquals(Version.SDK_NAME, result.getTaskCompleted().getSdkMetadata().getSdkName());
+    assertEquals(
+        Version.LIBRARY_VERSION, result.getTaskCompleted().getSdkMetadata().getSdkVersion());
+  }
+
+  @Test
+  public void setsSdkNameAndVersionIfDifferentFromHistory() throws Throwable {
```

Comment:
This test and the next I can't get to work because the setup in this suite isn't meant to work with more than one completion. I tried adjusting the mocks but it just led down dead end of the mock really not being suitable.

@Quinn-With-Two-Ns I can't find any tests that will allow me to really do what I want, which is just pass in some history and get a WFT response out that I can inspect, at this level. There are a bunch that operate at the workflow state machines level, but that's one level too deep. Any pointers would be welcome

Reply:
The tests in `io/temporal/internal/statemachines` take in a history and you can inspect the result. All the tests there are focused on a particular state machine, but no reason you can't add tests that focus on the workflow task state machine.

Reply:
Problem is this logic isn't in the workflow task state machine, it's in `ReplayWorkflowRunTaskHandler`, but, whatever. For the sake of making this easier to test with the existing stuff I'll just push it down and require one extra getter and just make sure it's set for completions with an integ test

Reply:
Ok, this is all fixed. The one remaining test at that level ensures the name/ver returned by state machines is actually attached to the response